lib/authentication/secure_storage/json_credential_manager.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2015-2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+const path = require('path');
+const Logger = require('../../logger');
+const fs = require('fs');
+const os = require('os');
+const Util = require('../../util');
+
+function JsonCredentialManager() {
+
+  this.userinfo = os.userInfo();
+
+  this.getTokenDir = function () {
+    let tokenDir = process.env.SF_TEMPORARY_CREDENTIAL_CACHE_DIR;
+
+    if (!Util.exists(tokenDir)) {
+      tokenDir = os.homedir();
+    } else {
+      Logger.getInstance().info(`The credential cache directory is configured by the user. The token will be saved at ${tokenDir}`);
+    }
+
+    if (!Util.exists(tokenDir)) {
+      Logger.getInstance().error(`Temporary credential cache directory is invalid, and the driver is unable to use the default location(home). 
+      Please assign the enviroment variable value SF_TEMPORARY_CREDENTIAL_CACHE_DIR to enable the default credential manager.`);
+      return null; 
+    }
+    const tokenCacheFile = path.join(tokenDir, 'temporary_credential.json');
+
+    if (fs.existsSync(tokenCacheFile)) {
+      try {
+        fs.accessSync(tokenCacheFile,  fs.constants.R_OK | fs.constants.W_OK);
```

Comment:
Our goal here was to verify that only user has permission to this file. What about using fs.constants.S_IRUSR and fs.constants.S_IWUSR

Reply:
In fs.access(Sync), fs.constants.S_IRUSR and fs.constants.S_IWUSR cannot be used because It is out of scope. This function only checks the permission, However, I changed the mode with fs.constants.S_IRUSR and fs.constants.S_IWUSR  when the driver writes or modifies the file.

Reply:
I will discuss it with security team.

Reply:
What about: 'fs.statSync('./yourfile.txt').mode'?

Reply:
Done.